.github/workflows/release.yaml

```diff
@@ -0,0 +1,280 @@
+        run: echo "${{ inputs.version }}" | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' || (echo "Version number is not formatted as vX.X.X" && false)
+
+  extract-version-details:
+    runs-on: ubuntu-latest
+    needs: verify-inputs
+    outputs:
+      full-version: ${{ steps.production.outputs.full-version || steps.unstable.outputs.full-version || steps.testing.outputs.full-version }}
+      aliased-version: ${{ steps.production.outputs.aliased-version || steps.unstable.outputs.aliased-version || steps.testing.outputs.aliased-version }}
+    steps:
+      - id: production
+        if: ${{ inputs.kind == 'production' }}
+        run: |
+          fullVersion="${{ inputs.version }}"
+          aliasedVersion=$(echo "${{ inputs.version }}" | sed -E 's/\.[0-9]+\.[0-9]+$//g')
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+      - id: unstable
+        if: ${{ inputs.kind == 'unstable' }}
+        run: |
+          fullVersion="unstable-${{ inputs.sha }}"
+          aliasedVersion="unstable"
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+      - id: testing
+        if: ${{ inputs.kind == 'testing' }}
+        run: |
+          fullVersion="testing-${{ inputs.sha }}"
+          aliasedVersion=""
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+
+  draft-full-version-release:
+    runs-on: ubuntu-latest
+    needs: extract-version-details
+    permissions:
+      contents: write
+      id-token: write
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.sha }}
+      - name: Ensure this release has not been published
+        run: gh release view ${{ needs.extract-version-details.outputs.full-version }} --json isDraft | jq -e '.isDraft == true'
+      - name: Push full version tags
+        run: |
+          git tag --force ${{ needs.extract-version-details.outputs.full-version }}
+          git push --tags --force origin ${{ needs.extract-version-details.outputs.full-version }}
+      - id: create-full-version-release
+        continue-on-error: true
+        name: Create the full version release
+        run: |
+          gh release create ${{ needs.extract-version-details.outputs.full-version }} \
+            --draft \
+            --prerelease=${{ inputs.kind != 'production' }} \
+            --generate-notes \
+            --title "Captain ${{ needs.extract-version-details.outputs.full-version }}"
+      - name: Update the release if it already exists
+        if: ${{ steps.create-full-version-release.outcome == 'failure' }}
+        run: gh release edit ${{ needs.extract-version-details.outputs.full-version }} --tag ${{ needs.extract-version-details.outputs.full-version }}
```

Comment:
💭 I don't think this is doing anything. If the tag moves the release should move with it (yesterday when I force pushed the v1 tag, the v1 release changed automatically)